src/cdk-experimental/menu/menu-item-selectable.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceBooleanProperty, BooleanInput} from '@angular/cdk/coercion';
+import {Input, Directive, Output, EventEmitter} from '@angular/core';
+import {CdkMenuItem} from './menu-item';
+
+/** Counter used to set a unique name for a selectable item */
+let _uniqueNameCounter = 0;
+/** Counter used to set a unique id for a selectable item */
+let _uniqueIdCounter = 0;
+
+/**
+ * Base class providing checked state for MenuItems along with outputting a clicked event when the
+ * element is triggered. It provides functionality for selectable elements.
+ */
+@Directive()
+export abstract class CdkMenuItemSelectable extends CdkMenuItem {
+  /** Event emitted when the selectable item is clicked */
+  @Output() clicked: EventEmitter<CdkMenuItemSelectable> = new EventEmitter();
+
+  /** Whether the element is checked */
+  @Input()
+  get checked() {
+    return this._checked;
+  }
+  set checked(value: boolean) {
+    this._checked = coerceBooleanProperty(value);
+  }
+  private _checked = false;
+
+  /** The name of the selectable element with a default value */
+  @Input() name: string = `cdk-selectable-item-${_uniqueNameCounter++}`;
+
+  /** The id of the selectable element with a default value */
+  @Input() id: string = `cdk-selectable-item-${_uniqueIdCounter++}`;
+
+  /** If the element is not disabled emit the click event */
+  trigger() {
+    if (!this.disabled) {
+      this.clicked.next(this);
+    }
+  }
+
+  static ngAcceptInputType_checked: BooleanInput;
```

Comment:
I've never seen a static class member used in TypeScript before. What is this for?

Reply:
when you do something like `<div disabled>` this is the equivalent of `disabled = ""` and therefore we need to coerce the truthy value. However we want checked to specify that it is a boolean type. This ensures that we can provide a truthy type but allows us to specify it as a boolean type.